src/query/interpreter.cpp

```diff
@@ -1018,10 +1018,26 @@ struct PullPlanVector {
   std::vector<std::vector<TypedValue>> values_;
 };
 
+struct TxTimeout {
+  TxTimeout() = default;
+  explicit TxTimeout(double value) noexcept : value_{std::in_place, value} {
+    // validation
+    // - negative timeout makes no sense
+    // - zero timeout means no timeout
+    if (value_ <= 0.0) value_.reset();
+  };
+  explicit operator bool() const { return value_.has_value(); }
+  double value() const { return *value_; }
```

Comment:
What if user sends value smaller than 0, then we do `reset` and on `value()` it will be garbage, right?

Reply:
Yes...the contract is that you will test before fetching. Should I rename to `valueUnchecked`, `valueUnsafe` or something else? (+ document that need to test before fetch)

Reply:
Maybe `valueUnsafe`? It sounds good to me

Reply:
fixed in latest commit